sdk/core/azure-core/CMakeLists.txt

```diff
@@ -120,6 +120,9 @@ target_include_directories(azure-core PUBLIC ${CURL_INCLUDE_DIRS})
 target_include_directories(azure-core INTERFACE ${nlohmann_json_INCLUDE_DIRS})
 target_link_libraries(azure-core INTERFACE CURL::libcurl Threads::Threads nlohmann_json::nlohmann_json)
 
+if(BUILD_TRANSPORT_CURL)
+  target_link_libraries(azure-core PRIVATE CURL::libcurl)
+endif()
```

Comment:
This change is also fixing https://github.com/Azure/azure-sdk-for-cpp/issues/926